common-service-core/src/main/java/com/ca/mfaas/message/core/Message.java

```diff
@@ -0,0 +1,167 @@
+                    Object[] messageParameters) {
+        this.requestedKey = requestedKey;
+        this.messageTemplate = messageTemplate;
+        this.messageParameters = messageParameters;
+    }
+
+    /**
+     * Return a {@link Message} object for the specified key after text and parameters validation.
+     *
+     * @param requestedKey the message key.
+     * @param messageTemplate the messageTemplate.
+     * @param messageParameters the object containing the message parameters.
+     * @return {@link Message}
+     */
+    public static Message of(String requestedKey,
+                             MessageTemplate messageTemplate,
+                             Object[] messageParameters) {
+
+        requireNotNull(requestedKey, "requestedKey can't be null");
+        requireNotNull(messageTemplate, "messageTemplate can't be null");
+        requireNotNull(messageParameters, "messageParameters can't be null");
+
+        validateMessageTextFormat(messageTemplate.getText(), messageParameters);
+        messageParameters = validateParameters(messageTemplate.getKey(), requestedKey, messageParameters);
+
+        return new Message(requestedKey, messageTemplate, messageParameters);
+    }
+
+
+    /**
+     * Checks that the specified object reference is not null and
+     * throws a {@link IllegalArgumentException} if it is.
+     *
+     * @param param the object reference to check for nullity
+     * @param message detail message to be used in the event
+     */
+    private static void requireNotNull(Object param, String message) {
+        if (param == null) {
+            throw new  IllegalArgumentException(message);
+        }
+    }
+
+    /**
+     * Validate the message text and parameters returning them as formatted String.
+     *
+     * @param messageText the message text.
+     * @param messageParameters the object containing the message parameters.
+     * @return a formatted String
+     */
+    private static String validateMessageTextFormat(String messageText, Object[] messageParameters) {
+        return String.format(messageText, messageParameters);
+    }
+
+    /**
+     * Check if the key is equal to the invalid key message. If it is, the parameter is set to the requested key,
+     * which will be displayed in the invalid key message text, otherwise the passed parameters are returned.
+     * @param messageKey the message key.
+     * @param requestedKey the message key.
+     * @param parameters the object containing the message parameters.
+     * @return an Object
+     */
+    private static Object[] validateParameters(String messageKey, String requestedKey, Object... parameters) {
+        if (messageKey.equals(Message.INVALID_KEY_MESSAGE)) {
+            return new Object[]{ requestedKey };
+        } else {
+            return parameters;
+        }
+    }
+
+    /**
+     * Converts the text after processing with {@link MessageTemplate} and Object[] message parameters by {@link Message} class.
```

Comment:
"...after processing " here should be some object. To process smth with smth else... Maybe you meant
"Converts the text after processing {@link MessageTemplate} and Object[] with the message parameters by {@link Message} class."?